### ISSUE TITLE
Timeout connecting to windows vEthernet from WSL2

### ISSUE THREAD

**Issue by Lite5h4dow** [microsoft/WSL#6526]
# Environment

```none
Microsoft Windows [Version 10.0.19041.746]
Linux version 5.4.72-microsoft-standard-WSL2 (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Oct 28 23:40:43 UTC 2020 (ArchWSL)
```

# Steps to reproduce

Following: https://docs.microsoft.com/en-us/windows/wsl/compare-versions#accessing-network-applications
I create a node server on windows on port 2999, and attempt to get a response via curl using the same steps illustrated in the link above.

#  Expected behavior
I would expect the response to appear as `hello world!` as illustrated.

# Actual behavior
I'm getting a timeout error using anything (curl, wget, axios, fetch)

**Comment by blake-mealey** [comment]
I have the same issue

From Window PowerShell, I run the following Node.js code:

```js
const express = require('express');

const app = express();
const port = process.env.PORT || 4312;

app.get('/', (req, res) => {
  res.send('Hello, world!');
});

app.listen(port, () => {
  console.log(`Listening on http://127.0.0.1:${port}`);
});
```

From Windows PowerShell I run `curl localhost:4312` and receive "Hello, world!"

From WSL2 Ubuntu 20.04 Bash I run `curl 172.23.144.1:4312` (I get the IP from `cat /etc/resolv.conf`) and receive "curl: (28) Failed to connect to 172.23.144.1 port 4312: Connection timed out"

**Comment by therealkenc** [comment]
Thank-you for the test case. Was able to reproduce here. Disabled firewall (which is a usual suspect) but no joy.

![image](https://user-images.githubusercontent.com/7049700/107425378-c161fc00-6ad3-11eb-9d28-1f8b5e94c33d.png)

**Comment by therealkenc** [comment]
Simpler repro that doesn't require installing Windows `node` . Using 445 (SMB/CIFS) as the Windows-side tcp4 listening socket (which is usually there and listening):

```
$ sudo apt install socat
$ WINHOST=$(ip route show default | awk '{print $3}')
$ echo $WINHOST
$ netstat.exe /an | grep 445 | grep LISTENING
$ socat TCP4:$WINHOST:445 -
```

![image](https://user-images.githubusercontent.com/7049700/108434389-6da29180-71fc-11eb-9be3-9cc5adf4f2ee.png)

There is nothing to learn from the `strace(1)` log, it is waiting on a `connect(2)`, natch:
```
541   socket(AF_INET, SOCK_STREAM, IPPROTO_TCP) = 5
541   recvfrom(3, 0x7ffe06b541d0, 519, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
541   fcntl(5, F_SETFD, FD_CLOEXEC)     = 0
541   recvfrom(3, 0x7ffe06b541d0, 519, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
541   recvfrom(3, 0x7ffe06b540e0, 519, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
541   connect(5, {sa_family=AF_INET, sin_port=htons(445), sin_addr=inet_addr("172.30.48.1")}, 16) = -1 ETIMEDOUT (Connection timed out)
541   recvfrom(3, 0x7ffe06b540e0, 519, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
541   getsockname(5, {sa_family=AF_INET, sin_port=htons(55852), sin_addr=inet_addr("0.0.0.0")}, [112->16]) = 0
541   openat(AT_FDCWD, "/etc/localtime", O_RDONLY|O_CLOEXEC) = 6
541   fstat(6, {st_mode=S_IFREG|0644, st_size=2892, ...}) = 0
541   fstat(6, {st_mode=S_IFREG|0644, st_size=2892, ...}) = 0
541   read(6, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\5\0\0\0\5\0\0\0\0"..., 4096) = 2892
541   lseek(6, -1838, SEEK_CUR)         = 1054
541   read(6, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\5\0\0\0\5\0\0\0\0"..., 4096) = 1838
541   close(6)                          = 0
541   getpid()                          = 541
541   write(2, "2021/02/18 15:19:16 socat[541] E"..., 92) = 92
541   shutdown(5, SHUT_RDWR)            = -1 ENOTCONN (Transport endpoint is not connected)
541   exit_group(1)                     = ?
541   +++ exited with 1 +++
```

At a loss as to why the connect doesn't succeed. Absent some externality I'm missing, it possibly be a regress, because checking whether a port can be `connect(2)ed` with `socat` (or `wget` or `curl` or whatever) is a fairly common exercise I've done in the past (natch). Full disclosure third-party networking software, [PIA](https://www.privateinternetaccess.com/) is installed but not live this session (no endorsement implied). That's probably not it. AV is Defender. Security policy controlled by a large corp. What is most puzzling is that there isn't more screeching, so there is something I am missing.

**Comment by ysctest** [comment]
Is this problem still unsolved?

**Comment by Lite5h4dow** [comment]
yes, this is still unsolved unfortunately

**Comment by ridai260** [comment]
Having this issue on both my WSL Ubuntu and Arch installations, exact same error. Works fine in CMD/Powershell. Guess I'm screwed?

**Comment by Lite5h4dow** [comment]
@ridai260 yeah :/

**Comment by Lite5h4dow** [comment]
> 

more like #7254 is a duplicate of this (look at the date) however this applies to any language and framework (including curl)

**Comment by aawilson** [comment]
The workaround in #7254 did nothing to resolve this for me (wsl kernel version 5.10.102.1), nor did @rickywu 's firewall rule.
